app/metal-controller-manager/internal/ipxe/ipxe_server.go

```diff
@@ -229,7 +229,9 @@ func newEnvironment(c client.Client, server *metalv1alpha1.Server) (env *metalv1
 		if err != nil {
 			return nil, err
 		}
-	default:
+	}
```

Comment:
this fixes a bug: `server.OwnerReferences` might be `!= nil`, but environment might not be available anyways, so we have to always fall back to the default environment.